clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -627,12 +822,38 @@ public Set<String> subscription() {
 
     @Override
     public void subscribe(Collection<String> topics) {
-        throw new KafkaException("method not implemented");
+        subscribe(topics, new NoOpConsumerRebalanceListener());
     }
 
     @Override
-    public void subscribe(Collection<String> topics, ConsumerRebalanceListener callback) {
-        throw new KafkaException("method not implemented");
+    public void subscribe(Collection<String> topics, ConsumerRebalanceListener listener) {
```

Comment:
Subscribe implies group rebalancing, I wonder if we should implement this later.